### PR TITLE
Experimental scheduler group abort functionality

### PR DESCRIPTION
**Pull request by chefsalim**
Adding some initial functionality to stop scheduled group jobs. This is currently primarily for development purposes, eventually there will be a more full fledged ability to stop/restart groups and jobs.

Signed-off-by: Salim Alam <salam@chef.io>

tenor-155592577

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/server.rs [hunk 1]
```diff
@@ -1103,6 +1103,30 @@ fn get_schedule(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+// TODO (SA) This is an experiemental dev-only function for now
+fn abort_schedule(req: &mut Request) -> IronResult<Response> {
+    let group_id = {
```
Comment (reset) [inline components/builder-depot/src/server.rs]: This should be an authenticated request even if it is experimental so people don’t cancel each other’s jobs :)

**Comment by reset** [comment]
This will be killer to get in!

File: components/builder-depot/src/server.rs (hunk 1 above)
Comment (chefsalim) [reply components/builder-depot/src/server.rs]: ah yeah - thanks for the catch!

Comment (chefsalim) [reply components/builder-depot/src/server.rs]: I've made it require  worker privilege for now - getting the right access check is one of the things that will need to be done right later.

**Comment by reset** [comment]
@chefsalim looks good to me!

**Pull request merged**